### PR TITLE
Add dark variant to OverflowMenu icon

### PR DESCRIPTION
**Pull request by tarikhamilton**
Before submitting a pull request, please make sure the following is done:

1. Make sure your updates are reflected in both the React and HTML side of MDL.
2. If you've fixed a bug or added code that should be tested, add tests!
3. Make sure the test suite passes (`npm test`).

### CODE REVIEW

File: components/SidebarNav/style.module.scss [hunk 1]
```diff
@@ -23,6 +23,7 @@ $submenu-width: 180px;
 .sidebar {
   position: fixed;
   top: 0;
+  left: 0;
```
Comment (josephfusco) [inline components/SidebarNav/style.module.scss]: Can we remove this unrelated change pls
```suggestion
```

File: components/OverflowMenu/react.js [hunk 2]
```diff
@@ -35,21 +36,27 @@ export class OverflowMenu extends Component {
 
   render() {
     const { displayOverflowMenuActions } = this.state
-    const { children } = this.props
+    const { children, variant } = this.props
     return (
-      <div className={cx('overflow-menu')}>
+      <div
+        className={cx('overflow-menu', {
+          'overflow-menu--dark': ['secondary', 'dark'].includes(variant),
```
Comment (josephfusco) [inline components/OverflowMenu/react.js]: @tarikhamilton should we account for this new prop in this component's tests? I believe we test for conditional classes elsewhere in MDL.

File: components/OverflowMenu/react.js [hunk 3]
```diff
@@ -35,21 +36,27 @@ export class OverflowMenu extends Component {
...
+        })}
+      >
         <button
           type="button"
           className={cx('overflow-menu__button')}
           onClick={this.toggleOverflowMenu}
         >
-          <svg
+          <Icon
```
Comment (josephfusco) [inline components/OverflowMenu/react.js]: Love the new Icon component 🔥

File: components/OverflowMenu/react.js (hunk 2 above)
Comment (josephfusco) [inline components/OverflowMenu/react.js]: @tarikhamilton @simonwebdesign I'm not too familiar with how this component is supposed to work so these concerns may not be relevant however I do have a couple general questions regarding this new variant prop:

1. Do you think we would ever want the overflow menu content to also use the variant styles and if so what does that look it? i.e. dark variant button and dark style for the overflow menu.
2. If yes to the above, would we ever need any of the following variants?
    - dark button with light popup
    - light button with dark popup
    If so, we may want to consider updating the prop names to be more specific since dark seems like both the button & overflow menu.

File: components/SidebarNav/style.module.scss (hunk 1 above)
Comment (tarikhamilton) [reply components/SidebarNav/style.module.scss]: It is needed. If there is padding in the wrapper (aka Storybook), if `left` isn't specified, it'll be lined up with the padding, creating a little gap.

File: components/OverflowMenu/react.js (hunk 2 above)
Comment (tarikhamilton) [reply components/OverflowMenu/react.js]: I don't think it is totally necessary.

Comment (tarikhamilton) [reply components/OverflowMenu/react.js]: We do want the menu to be dark too, but it isn't designed yet. I was waiting for Simon to do it, but we can always update the library incrementally.

File: components/SidebarNav/style.module.scss (hunk 1 above)
Comment (josephfusco) [reply components/SidebarNav/style.module.scss]: I see. Was that meant to be in this PR? https://github.com/masonitedoors/MDL/pull/102
This PR was just adding a variant to the OverflowMenu component.

File: components/OverflowMenu/react.js (hunk 2 above)
Comment (josephfusco) [reply components/OverflowMenu/react.js]: While I would agree, just wanted to call it out for project consistency. We may want to discuss testing expectations in our next MDL meeting.

File: components/SidebarNav/style.module.scss (hunk 1 above)
Comment (tarikhamilton) [reply components/SidebarNav/style.module.scss]: yea

**Comment by simonwebdesign** [comment]
Added guidance in Zeplin.

Main navigation: 
zpl://components?stid=5d0a60e44725312d15a183b6&coids=5e67a638de2abb14345629ce

Overflow menu:
zpl://components?stid=5d0a60e44725312d15a183b6&coids=5e6f7a5d82a344183b538649

File: components/SidebarNav/style.module.scss (hunk 1 above)
Comment (josephfusco) [reply components/SidebarNav/style.module.scss]: I think it's fine here just wanted to call it out 👍this looks good though.
